tree/30_generic_methods/variable_string_from_augeas.cf

```diff
@@ -0,0 +1,88 @@
+#####################################################################################
+# Copyright 2020 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+#
+# @name Variable string from Augeas 
+# @description Use Augeas binary to call Augtool command and get a node's label value.
+# @documentation There is two ways you can use this method, either you simply provide a path to the node label as a parameter or you can specify a file associated with a lens then the regular path. 
+# The generic method will get a node's label vaule otherwise if the augeas isn't installed on the agent, it will produces an error.
+#
+# #### Two uses cases examples:
```

Comment:
Could you add an explanation about the difference of behavior when providing file and lens or not?

Reply:
sure!